core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -383,6 +384,16 @@ public FileInfo getFileInfo(AlluxioURI path)
       mPermissionChecker.checkPermission(FileSystemAction.READ, path);
       // getFileInfo should load from ufs if the file does not exist
       getFileId(path);
+
+      if (!mInodeTree.inodePathExists(path)) {
+        try {
+          loadMetadata(path, LoadMetadataOptions.defaults().setCreateAncestors(true));
```

Comment:
hmm, I don't think this is true. By default, getFileInfo doesn't load since mLoadDirectChildren is by default = false